### DISCUSSION TITLE
📖 등재할 기술 용어 모음

### DISCUSSION THREAD

**Discussion by hochan222** [ko-translation-team/glossary#13] [일반]
기술 용어 안내서 작성에 앞서서 **논란의 여지가 없는 보편적인 기술 용어**와 **토론이 필요한 기술 용어**를 모으고자 합니다. 댓글로 추가되면 좋을 단어 부탁드립니다!

## 논란의 여지가 없는 보편적인 기술 용어

`배열(Array)`, `오류(Error)`와 같이 보편적으로 받아들여져서 토론할 필요가 없는 용어들을 먼저 모아서 기술 용어 안내서를 작성하고자 합니다.

아래 예시와 같이 추가 되면 좋을 단어들 알려주세요!

| 원어 | 번역어 |
| ---- | ---- |
| Array | 배열 |
| Error | 오류 |

## 토론이 필요한 기술 용어

사유와 사용 예시도 함께 작성해주세요!

**Comment by hochan222** [comment]
## 논란의 여지가 없는 보편적인 기술 용어

**Reply by EatChangmyeong** [reply]
https://en.wikipedia.org/wiki/Glossary_of_computer_science 에서 뽑아올 게 많아 보이네요 시간 관계상 음차 아닌 것 위주로 처음 몇 개만 적어볼게요 😅

- artificial intelligence = 인공지능
- concurrency = 동시성
- conditional = 조건-
- data structure = 자료구조
- declaration = 선언
- exception handling = 예외 처리
- expression = 표현식, (합성어로서) -식
- inheritance = 상속
- input/output = 입력/출력, 입출력
- ...

**Reply by RanolP** [reply]
- data type = 자료형
- sign = 부호
- function = 함수
- bandwidth = 대역폭
- binary number = 이진수
- boolean algebra = 부울 대수

**Reply by roo0t** [reply]
* abstraction = 추상화
* augmented reality (AR) = 증강현실
* synchronous / asynchronous = 동기 / 비동기
* statement = 문(文)
* develop / development / developer = 개발하다 / 개발 / 개발자
* optimization = 최적화
* integer = 정수
* language = 언어
* object = 객체
* object-oriented: 객체지향(적)
* (unary/binary/ternary) operator = (단항/이항/삼항) 연산자
* polymorphism = 다형성
* string = 문자열
* value = 값
* variable = 변수
* path = 경로
* environment variable = 환경 변수
* explorer = 탐색기
* user = 사용자
* asymptotic = 점근(적)
* sort = 정렬(하다)
* bipartite graph = 이분 그래프
* parent/child = 부모/자식

**Reply by SnowMarble** [reply]
안녕하세요, 몇몇 분 께서 벌써 적지 않은 단어를 정리해주셨지만, 갑자기 드는게 "논란의 여지가 없는 보편적 기술 용어"를 정말 **정리**해 할까 였습니다. "보편적"이라는 말 처럼 어느정도의 IT 지식을 가지고 있을 번역 기여자가 이 단어집을 참고하는 경우가 단어 정리에 드는 노력에 비해 적지 않을까라고 생각되었습니다. `function` 이 `함수`인지 확인하지 않는것 처럼 말이죠.

또한 보편적 이라는 모호한 기준과 사전을 통해서도 충분히 뜻을 알 수 있는 단어가 무분별하게 추가되어 용어집이 기여자들을 압도할 수 있겠다고도 느꼈습니다.

보편적 용어에 대한 취지를 이해하지 못하고 하는 말일수도 있지만, 만약 보편적 단어를 정리해야 하는 상황이라면 최소한 "보편적"의 기준이 필요해보입니다. :)

**Reply by hochan222** [reply]
확실히 논란이 있을만한 단어들에 대해서만 토론의 장을 열고 기록을 남기는것이 효율적일 수 있을것같네요.

**Comment by hochan222** [comment]
## 토론이 필요한 기술 용어

**Reply by SnowMarble** [reply]
저번에 얘기하다 결론을 짓지 못 했던 것 같습니다. [#10198](https://github.com/mdn/translated-content/issues/10198)
* margin
* padding

**Reply by roo0t** [reply]
scalable, extensible

참고: https://occamsrazr.net/tt/3
